backend/frege/indexers/utils/gitlab.py

```diff
@@ -83,12 +86,19 @@ def _projects(self):
             next_page = projects_response.links.get('next', {}).get('url')
             yield projects_response.json()
 
-    def _commit_hash(self, project_id: int):
+    def _commit_hash(self, project_id):
         """Gets the latest commit hash in the default branch"""
-        project_commits = self._get(f"{BASE_ENDPOINT}/{project_id}/repository/commits")
-        try:
-            return project_commits.json()[0]['id']
-        except KeyError:
+
+        response = self._get(f"{BASE_ENDPOINT}/{project_id}/repository/commits")
+
+        if response.status_code != 200:
+            logger.warning(f"Unable to fetch commits for project {project_id}, status code: {response.status_code}")
             return None
 
+        commits = response.json()
+        if not commits:
+            logger.warning(f"No commits found for project {project_id}")
+            
+            return None
 
+        return commits[0]['id']
```

Comment:
If I understand correctly:

```suggestion
				mainBranch = commits[0]
				hash = mainBranch['id']
        return hash
```

Reply:
done